### PR TITLE
jira/HZN-460: Enhance the heatmap to use Services

### PR DESCRIPTION
**Pull request by christianpape**
Please review and merge:

JIRA: http://issues.opennms.org/browse/HZN-460
BAMBOO: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS173

### CODE REVIEW

File: opennms-doc/guide-admin/src/asciidoc/text/webui/heatmap/introduction.adoc [hunk 1]
```diff
@@ -21,7 +21,8 @@ image:webui/heatmap/heatmap.png[]
                                                            default.
                                                                                                                         | `alarms`
 | `org.opennms.heatmap.defaultHeatmap`       | _String_  | This option defines which _Heatmap_ is displayed by default.
-                                                           The two valid options are `categories` or `foreignSources`   | `categories`
+                                                           The three valid options are `categories`, `foreignSources` and
+                                                           'services'.                                                  | `categories`
```
Comment (mvrueden) [inline opennms-doc/guide-admin/src/asciidoc/text/webui/heatmap/introduction.adoc]: Maybe just write `Valid options are: ...` So you do not have to change the numbers everytime the list changes.

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/HeatMapRestService.java [hunk 2]
```diff
@@ -51,7 +39,17 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
```
Comment (mvrueden) [inline opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/HeatMapRestService.java]: Please configure your IDE to match with the ecilpse import pattern to avoid unnecessary changes.

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/HeatMapRestService.java [hunk 3]
```diff
@@ -168,6 +171,16 @@ public Response outagesByForeignsources() throws IOException {
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @Transactional
+    @Path("outages/managedServices")
+    public Response outagesByServices() throws IOException {
+        final List<HeatMapElement> heatMapElements = m_outageDao.getHeatMapItemsForEntity("service.servicename", "service.serviceid", null, null);
+        final JSONObject jo = new JSONObject(transformResults(heatMapElements, System.getProperty(SERVICE_FILTER_PROPERTY_KEY, SERVICE_FILTER_PROPERTY_DEFAULT)));
+        return Response.ok(jo.toString(), MediaType.APPLICATION_JSON).build();
```
Comment (mvrueden) [inline opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/HeatMapRestService.java]: Do you have the same issue here as the one Dustin yesterday fixed?
Does `jo.toString()` raise an exception?

File: opennms-webapp/src/main/webapp/heatmap/heatmap-box.jsp [hunk 4]
```diff
@@ -88,11 +103,15 @@
     if ("categories".equals(heatmap)) {
         title += " (by Categories)";
     }
+
+    if ("managedServices".equals(heatmap)) {
+        title += " (by Services)";
+    }
 %>
```
Comment (mvrueden) [inline opennms-webapp/src/main/webapp/heatmap/heatmap-box.jsp]: all of this logic should live in an controller instead of the jsp

Comment (christianpape) [inline opennms-webapp/src/main/webapp/heatmap/heatmap-box.jsp]: This is only the title of the heatmap, so no complicated logic.

File: opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/HeatMapRestService.java (hunk 3 above)
Comment (christianpape) [inline opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/HeatMapRestService.java]: No, extremly absolute total sowasvon unrelated!

**Comment by mvrueden** [comment]
From my point of view everything is awesome, except the commit messages. Please rephrase so that the issue number occurs: `HZN-460: Reorganized imports.`, etc.

You can achieve that by `git rebase <commit> -i -p` and change from `squash` to `reword (r)`
